components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +264,136 @@ func NewAdxl345(
 			}
 		}
 	})
+	sensor.interruptsFound = make(map[InterruptID]int)
+	if err := sensor.readInterrupts(sensor.cancelContext); err != nil {
+		// shut down goroutine reading sensor in the background
+		sensor.cancelFunc()
+		return nil, err
+	}
+	if err := sensor.configureInterruptRegisters(ctx, interruptConfigurations[IntMapAddr]); err != nil {
+		// shut down goroutine reading sensor in the background
+		sensor.cancelFunc()
+		return nil, err
+	}
+
+	interruptMap := map[string]board.DigitalInterrupt{}
+	if (cfg.SingleTap != nil) && (cfg.SingleTap.InterruptPin != "") {
+		interruptMap, err = addInterruptPin(b, cfg.SingleTap.InterruptPin, interruptMap)
+		if err != nil {
+			// shut down goroutine reading sensor in the background
+			sensor.cancelFunc()
+			return nil, err
+		}
+	}
+
+	if (cfg.FreeFall != nil) && (cfg.FreeFall.InterruptPin != "") {
```

Comment:
Didn't we already check the second half of this in the config validation?

Reply:
Yes, but if we get here and `cfg.FreeFall` has not been configured and we try to set these values, won't it cause a panic?